### PR TITLE
core: Add network test

### PR DESCRIPTION
**Pull request by patrickelectric**
docker: patrickelectric/companion-core:network_test

Helps #448

### CODE REVIEW

File: core/frontend/src/views/NetworkTestView.vue [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+      speed: 0,
+      download_speed: undefined as number | undefined,
+      upload_speed: undefined as number | undefined,
+      latency_us: undefined as number | undefined,
+      websocket: undefined as WebSocket | undefined,
+      interval: 0,
+    }
+  },
+  beforeDestroy() {
+    clearInterval(this.interval)
```
Comment (Williangalvani) [inline core/frontend/src/views/NetworkTestView.vue:112]: you should close the websocket here, too.

Comment (patrickelectric) [reply core/frontend/src/views/NetworkTestView.vue:112]: Done!

File: core/frontend/src/views/NetworkTestView.vue [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+      await back_axios({
+        method: 'get',
+        url: '/network-test/get_file',
+        timeout: 20000,
+        data: {
+          size: one_hundred_mega_bytes,
+          avoid_cache: new Date().getTime(),
+        },
+        onDownloadProgress: (progress_event) => {
+          const seconds = (new Date().getTime() - start_time) / 1000
```
Comment (rafaellehmkuhl) [inline core/frontend/src/views/NetworkTestView.vue:207]: `date-fns` (which we already use) has some cool time helpers:

```javascript
start_date = new Date()
const seconds = differenceInSeconds(new Date(), start_date)
```

File: core/tools/nginx/nginx.conf [hunk 3]
```diff
@@ -119,6 +119,15 @@ http {
             proxy_pass http://127.0.0.1:6020;
```
Comment (rafaellehmkuhl) [inline core/tools/nginx/nginx.conf:119]: typo on commit name ("endpoint")

**Review by rafaellehmkuhl — CHANGES_REQUESTED** [review]

File: core/frontend/src/views/NetworkTestView.vue (hunk 2 above)
Comment (patrickelectric) [reply core/frontend/src/views/NetworkTestView.vue:207]: Nice!

File: core/tools/nginx/nginx.conf (hunk 3 above)
Comment (patrickelectric) [reply core/tools/nginx/nginx.conf:119]: Done.

File: core/frontend/src/views/NetworkTestView.vue [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+import notifications from '@/store/notifications'
+import { network_speed_test_service } from '@/types/frontend_services'
+import back_axios from '@/utils/api'
+
+enum State {
+  None = '...',
+  DownloadSpeed = 'Download speed',
+  UploadSpeed = 'Upload speed',
+  Done = 'Done',
```
Comment (rafaellehmkuhl) [inline core/frontend/src/views/NetworkTestView.vue:93]: 1. Click to test download/upload speed
2. Testing download speed
3. Testing upload speed
4. Test done

**Review by rafaellehmkuhl — CHANGES_REQUESTED** [review]

File: core/frontend/src/views/NetworkTestView.vue [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+            color="blue"
+            size="200"
+          >
+            {{ speed.toFixed(2) }} MB/s
+          </v-progress-circular>
+
+          <v-list>
+            <v-list-item>
+              <v-list-item-icon>
+                <v-icon>
```
Comment (rafaellehmkuhl) [inline core/frontend/src/views/NetworkTestView.vue:30]: `v-tooltip="'Download speed'"`

File: core/frontend/src/views/NetworkTestView.vue [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+                </v-icon>
+              </v-list-item-icon>
+              <v-list-item-subtitle>
+                {{ download_speed ? download_speed.toFixed(2) : '...' }} MB/s
+              </v-list-item-subtitle>
+            </v-list-item>
+
+            <v-list-item>
+              <v-list-item-icon>
+                <v-icon>
```
Comment (rafaellehmkuhl) [inline core/frontend/src/views/NetworkTestView.vue:41]: `v-tooltip="'Upload speed'"`

File: core/frontend/src/views/NetworkTestView.vue [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+                </v-icon>
+              </v-list-item-icon>
+              <v-list-item-subtitle>
+                {{ upload_speed ? upload_speed.toFixed(2) : '...' }} MB/s
+              </v-list-item-subtitle>
+            </v-list-item>
+
+            <v-list-item>
+              <v-list-item-icon>
+                <v-icon>
```
Comment (rafaellehmkuhl) [inline core/frontend/src/views/NetworkTestView.vue:52]: `v-tooltip="'Latency'"`

**Review by rafaellehmkuhl — CHANGES_REQUESTED** [review]

**Review by rafaellehmkuhl — APPROVED** [review]

**Pull request merged**